src/app/containers/Ad/index.test.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
+import { RequestContextProvider } from '#contexts/RequestContext';
+import { ServiceContextProvider } from '#contexts/ServiceContext';
+import { ToggleContext } from '#contexts/ToggleContext';
+
+import AdContainer from './index';
+
+const defaultToggleState = {
+  local: {
+    ads: {
+      enabled: true,
+    },
+  },
+  test: {
+    ads: {
+      enabled: true,
+    },
+  },
+  stage: {
+    ads: {
+      enabled: false,
+    },
+  },
+  live: {
+    ads: {
+      enabled: false,
+    },
+  },
+};
+
+const mockToggleDispatch = jest.fn();
+
+const toggleContextMock = {
+  toggleState: defaultToggleState,
+  toggleDispatch: mockToggleDispatch,
+};
+
+describe('Ad Container', () => {
+  shouldMatchSnapshot(
+    'should correctly render an amp leaderboard ad',
```

Comment:
- Add test for services where ads are not expected in order to make sure that ads won't randomly render where we don't want them.
- Add test for .amp page and not just the container. (Just check that on an amp page, an amp ad is rendered correctly)

Reply:
Tests added